new_actor.py

```diff
@@ -0,0 +1,262 @@
+        dataC = file.read()
+        dataC = dataC.replace("{actorSpec}", actorSpec)
+        dataC = dataC.replace("{actorDefine}", actorDefine)
+        dataC = dataC.replace("{actorFileName}", actorFileName)
+        dataC = dataC.replace("{objectDefine}", objectDefine if createObject else "OBJECT_GAMEPLAY_KEEP") 
+        dataC = dataC.replace("{actorVar}", "ActorProfile" if useActorProfile else "ActorInit")
+        dataC = dataC.replace("{actorInitVar}", "Profile" if useActorProfile else "InitVars")
+    
+    with open(f"{filePathActor}/{actorFileName}.c", 'w') as file: 
+        file.write(dataC) 
+    
+    with open(f"{filePathActor}/{actorFileName}.h", 'r') as file:
+        dataH = file.read()
+        dataH = dataH.replace("{actorSpec}", actorSpec)
+        dataH = dataH.replace("{actorFileNameCaps}", actorFileName.upper())
+        dataH = dataH.replace("{includeObject}", f'\n#include "assets/objects/{objectSpec}/{objectSpec}.h"' if createObject else "")
+    
+    with open(f"{filePathActor}/{actorFileName}.h", 'w') as file: 
+        file.write(dataH) 
+
+    if createObject:
+        # OBJECT
+        shutil.copyfile(Path(path.curdir, "template_files/object.c"), f"{filePathObject}/{objectSpec}.c")
+        shutil.copyfile(Path(path.curdir, "template_files/object.h"), f"{filePathObject}/{objectSpec}.h")
+
+        with open(f"{filePathObject}/{objectSpec}.c", 'r') as file:
+            dataC = file.read()
+            dataC = dataC.replace("{objectSpec}", objectSpec) 
+        
+        with open(f"{filePathObject}/{objectSpec}.c", 'w') as file: 
+            file.write(dataC) 
+        
+        with open(f"{filePathObject}/{objectSpec}.h", 'r') as file:
+            dataH = file.read()
+            dataH = dataH.replace("{objectSpec}", objectSpec) 
+            dataH = dataH.replace("{objectSpecCap}", objectSpec.upper())  
+        
+        with open(f"{filePathObject}/{objectSpec}.h", 'w') as file: 
+            file.write(dataH) 
+          
+def main():
+    names = convertActorName(actorName)
+
+    print(f'Creating new Actor: {names["actorSpec"]}')
+
+    makeFilesAndFolders(names["actorSpec"], names["actorFileName"], names["objectSpec"])
+    actorFileLine = addToTables(names["actorSpec"], names["actorDefine"], names["objectSpec"], names["objectDefine"])
+    addToSpec(names["actorSpec"], names["actorFileName"], names["objectSpec"], actorFileLine)
+    completeFiles(names["actorSpec"], names["actorDefine"], names["actorFileName"], names["objectSpec"], names["objectDefine"])
+
+    print(f'{names["actorSpec"]} created!')
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-name", required = True, help = "Name of actor. Use alphanumeric and/or _ as valid characters", default = "")
+    parser.add_argument("--noobject", required = False, help = "Optional. Use if you do not want to create an object for this actor", action='store_false')
```

Comment:
```suggestion
    parser.add_argument("-n", "--name", required = True, help = "Name of actor. Use alphanumeric and/or _ as valid characters", default = "")
    parser.add_argument("--noobject", required = False, help = "Optional. Use if you do not want to create an object for this actor", action='store_false')
```
not a fan of the inconsistency between ``-name`` and ``--noobject``

I'm not sure how people are using this but I'd invert the ``--noobject`` option so you could do ``./new_actor.py -n test -o`` to tell you want the object (I can't find a short name for "no object")